### PR TITLE
Amir's SleepBuddy2 project

### PR DESCRIPTION
**Pull request by LiricTech**
---
name: "@LiricTech"
project: "The SleepBuddy"
---

SleepBuddy's logo

# The SleepBuddy

## 💁‍♂️ Summary:

The SleepBuddy is a smart clock ⏰ that helps you sleep. It puts some calm music 🎵 and a night light 🔅 when you go to bed. The night light brightness and the music volume turns down as you fall asleep.

The SleepBuddy can also :

- Give the room temperature 🌡️
- Turn on light (night light) when it detects something moving 🐕🐈🚶
- Display 👀 the music you are listening to and the time thanks to its screens.
- Share information about your sleep with your friend's SleepBuddy (cross project with @CodingForAddicts) thanks to its wifi 🛜 antenna.

## ✅ Conditions to submit proposal:

- [x] Be a current high school student 🧑‍🎓 (or younger)
- [x] Submit an RSVP at https://hackclub.com/winter
- [x] Be a member of the Hack Club Slack (https://hackclub.com/slack)
- [x] Review my project proposal

## 📃 Plan:
1. Buy the hardware 💰
2. Print the prototype of the box in a local FabLab
3. Create the final box in polypropylene and wood
4. Program 💻 my arduino with all the components (uses in the summary)
5. Put all the components together in the box

## 💸 Budget:

| Product   | Supplier/Link  | Cost (including some taxes) |
|:--|:--|:--|
| Arduino UNO REV 2 WIFI/Bluetooth | [arduino](https://store.arduino.cc/products/arduino-uno-wifi-rev2?variant=35570905743511) | 55,90 $ (there is a promo code so I don't know the exact price) |
| Stereo Enclosed Speaker - 3W 8Ω (x2) | [arduino](https://store.arduino.cc/products/stereo-enclosed-speaker-3w-8ω?variant=40119999103127) | 9,60 $ (there is a promo code so I don't know the exact price) |
| NeoPixel LED strip - 30 RGB LEDs, 0,5m | [semageek](https://boutique.semageek.com/fr/1400-bande-de-led-neopixel-30-led-rgb-avec-connecteur-jst-05m-3007283931097.html?search_query=Bande+de+LED+NeoPixel+30+LED+RGB+avec+connecteur+JST+-+0%2C5m&results=355) | 15,93 $ |
| Humidity and temperature sensor | [semageek](https://boutique.semageek.com/fr/138-dht11-capteur-de-temperature-et-humidite-digital-3004931765998.html?search_query=temperature&results=65) | 7,00 $ |
| Grove - Touch Sensor (button on the top to stop the alarm) | [arduino](https://store.arduino.cc/products/grove-touch-sensor?variant=35573916991639) | 8,07 $ |
| SUBTOTAL (arduino and semageek products + shipping cost)  |  | 107,64 $ |
| Switch on-off (turn on and off the SleepBuddy) | [amazon](https://www.amazon.fr/ARTGEAR-Interrupteur-Bascule-Positions-Broches/dp/B07GDCNXKP/ref=sr_1_21?crid=D4O86S68WC7U&keywords=switch%2Barduino&qid=1671830001&sprefix=switch%2Barduin%2Caps%2C180&sr=8-21&th=1) | 10,61 $ |
| Movement sensor | [amazon](https://www.amazon.fr/AZDelivery-HC-SR501-ultrasons-T%C3%A9l%C3%A9m%C3%A8tre-Raspberry/dp/B07CNBYRQ7/ref=sr_1_2?__mk_fr_FR=%C3%85M%C3%85%C5%BD%C3%95%C3%91&crid=23F3OXHPOD84&keywords=movement%2Bsensor%2Barduino&qid=1671830580&sprefix=movement%2Bsensor%2Barduino%2Caps%2C87&sr=8-2&th=1) | 14,32 $ |
| Jumper Wires | [amazon](https://www.amazon.fr/YXPCARS-câbles-jumper-femelle-Arduino/dp/B08HQ7K6M7/ref=sr_1_5?crid=TV7B8USTG35U&keywords=jumper+cable&qid=1671636526&sprefix=jumper+ca%2Caps%2C108&sr=8-5) | 11,66 $ |
| wood and Polypropylene parts | Local FabLab (idk which one exactly) | max 50 $ |
| AZDelivery Display Module Vert LCD |  [amazon](https://www.amazon.fr/azdelivery-HD44780-1602-Module-écran-16-caractères-Arduino/dp/B01N3B8JMN/ref=sr_1_5?keywords=ecran%2Blcd%2Barduino&qid=1671636943&sr=8-5&th=1) | 6,89 $  |
| AZDelivery LED Display |  [amazon](https://www.amazon.fr/AZDelivery-Digital-Display-Arduino-Raspberry/dp/B06X952QXS/ref=sr_1_11?__mk_fr_FR=ÅMÅŽÕÑ&crid=FRXJMXVZ4D40&keywords=segmented%2Bi2c%2Bdisplay%2Barduino&qid=1671637038&sprefix=segmented%2Bi2c%2Bdispal%2Barduino%2Caps%2C82&sr=8-11&th=1) | 6,89 $ |
| Breadboard | [amazon](https://www.amazon.fr/gp/product/B08R2YM6RK/ref=ox_sc_act_title_1?smid=A2FA71JYWLOOYW&psc=1) | 6,27 $|
| TOTAL including taxes + shipping costs |  |  214,28 $ (depends on the price of some components) |

### CODE REVIEW

File: Amir's SleepBuddy project.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+- [x] Be a current high school student 🧑‍🎓 (or younger)
+- [x] Submit an RSVP at https://hackclub.com/winter
+- [x] Be a member of the Hack Club Slack (https://hackclub.com/slack)
+- [x] Review my project proposal
+
+## 📃 Plan:
+1. Buy the hardware 💰
+2. Print the prototype of the box in a local FabLab
+3. Create the final box in polypropylene and wood
+4. Program 💻 my arduino with all the components (uses in the summary)
```
Comment (exu3) [inline Amir's SleepBuddy project.md:33]: Could you list out the components and their uses either here, or in the summary? Currently, you just have the uses listed out in the summary but not the components. (eg. use a temperature/humidity sensor to take the room temperature)

File: Amir's SleepBuddy project.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+| Grove - Touch Sensor (button on the top to stop the alarm) | [arduino](https://store.arduino.cc/products/grove-touch-sensor?variant=35573916991639) | 8,07 $ |
+| SUBTOTAL (arduino and semageek products + shipping cost)  |  | 107,64 $ |
+| Switch on-off (turn on and off the SleepBuddy) | [amazon](https://www.amazon.fr/ARTGEAR-Interrupteur-Bascule-Positions-Broches/dp/B07GDCNXKP/ref=sr_1_21?crid=D4O86S68WC7U&keywords=switch%2Barduino&qid=1671830001&sprefix=switch%2Barduin%2Caps%2C180&sr=8-21&th=1) | 10,61 $ |
+| Movement sensor | [amazon](https://www.amazon.fr/AZDelivery-HC-SR501-ultrasons-T%C3%A9l%C3%A9m%C3%A8tre-Raspberry/dp/B07CNBYRQ7/ref=sr_1_2?__mk_fr_FR=%C3%85M%C3%85%C5%BD%C3%95%C3%91&crid=23F3OXHPOD84&keywords=movement%2Bsensor%2Barduino&qid=1671830580&sprefix=movement%2Bsensor%2Barduino%2Caps%2C87&sr=8-2&th=1) | 14,32 $ |
+| Jumper Wires | [amazon](https://www.amazon.fr/YXPCARS-câbles-jumper-femelle-Arduino/dp/B08HQ7K6M7/ref=sr_1_5?crid=TV7B8USTG35U&keywords=jumper+cable&qid=1671636526&sprefix=jumper+ca%2Caps%2C108&sr=8-5) | 11,66 $ |
+| wood and Polypropylene parts | Local FabLab (idk which one exactly) | max 50 $ |
```
Comment (exu3) [inline Amir's SleepBuddy project.md:49]: > (idk which one exactly)

Before we approve your project, could you decide which local FabLab you might use?

**Review by exu3 — CHANGES_REQUESTED** [review]
@LiricTech this project sounds neat. I like how it's interactive/collaborative in a way that you can share with your friends.

Your proposal is looking great. I left a few comments below. If you could address those and change the title of your document to `LiricTech.md`, I'll give it another review!

File: Amir's SleepBuddy project.md (hunk 1 above)
Comment (LiricTech) [reply Amir's SleepBuddy project.md]: I've added the components as explanations to the SleepBuddy's fonctions in the summary. Let me know if it's ok.

Comment (exu3) [reply Amir's SleepBuddy project.md]: Looks great, thanks!

**Review by exu3 — APPROVED** [review]
Looks great @LiricTech, thanks for making the changes!

A member of the @hackclub/bank-ops team will follow up with next steps on receiving the grant through Hack Club Bank.

**Review by MelSmith104 — COMMENTED** [review]
hey @LiricTech ! i love love love your proposal! your project sounds great and so useful to have! im one of those people that needs some sort of noise to fall asleep to so i 100% get the benefits of having something like this! im dm'ing you on slack right now so that we can chat about next steps!

**Comment by MelSmith104** [comment]
hi! just realized that you're already set up on hack club bank with @CodingForAddicts :) can you please send me your slack id so that we can chat about next steps?

**Comment by CodingForAddicts** [comment]
Hey! He’s in my team because I invited him in demo mode :) but we are making two linked projects (we are not ONE project).

**Comment by LiricTech** [comment]
Hello ! My Slack ID is @Amir but I may not be disponible right now :-/

**Comment by MelSmith104** [comment]
hey @CodingForAddicts thank you for the clarification! that's super helpful! @LiricTech looks like there are multiple slack users with the same slack name - can you please ping me @mel so that we can set up next steps :)

**Review by MelSmith104 — APPROVED** [review]

**Pull request merged**